pkg/cloudprovider/aws/api.go

```diff
@@ -0,0 +1,165 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package aws
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/ec2metadata"
+	"github.com/aws/aws-sdk-go/aws/endpoints"
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/ssm"
+	"github.com/awslabs/karpenter/pkg/apis/provisioning/v1alpha1"
+	"github.com/awslabs/karpenter/pkg/cloudprovider"
+	"github.com/awslabs/karpenter/pkg/cloudprovider/aws/utils"
+	"github.com/awslabs/karpenter/pkg/utils/functional"
+	"github.com/awslabs/karpenter/pkg/utils/project"
+	"github.com/patrickmn/go-cache"
+	"go.uber.org/zap"
+	v1 "k8s.io/api/core/v1"
+)
+
+const (
+	// CacheTTL restricts QPS to AWS APIs to this interval for verifying setup resources.
+	CacheTTL = 5 * time.Minute
+	// CacheCleanupInterval triggers cache cleanup (lazy eviction) at this interval.
+	CacheCleanupInterval = 10 * time.Minute
+	// ClusterTagKeyFormat is set on all Kubernetes owned resources.
+	ClusterTagKeyFormat = "kubernetes.io/cluster/%s"
+	// KarpenterTagKeyFormat is set on all Karpenter owned resources.
+	KarpenterTagKeyFormat = "karpenter.sh/cluster/%s"
+)
+
+var (
+	SupportedOperatingSystems = []string{
+		v1alpha1.OperatingSystemLinux,
+	}
+	SupportedArchitectures = []string{
+		v1alpha1.ArchitectureAmd64,
+		v1alpha1.ArchitectureArm64,
+	}
+)
+
+type API struct {
```

Comment:
IMO you should keep this concept named "cloud provider" since it's so common across the k8s ecosystem.